criu/proc_parse.c

```diff
@@ -616,10 +621,29 @@ static int handle_vma(pid_t pid, struct vma_area *vma_area, const char *file_pat
 			pr_info("path: %s\n", file_path);
 			vma_area->e->status |= VMA_AREA_SYSVIPC;
 		} else {
-			if (is_anon_shmem_map(st_buf->st_dev)) {
+			/* Dump shmem dev, hugetlb dev (private and share) mappings the same way as memfd
+			 * when possible.
+			 */
+			if (is_memfd(st_buf->st_dev) || is_anon_shmem_map(st_buf->st_dev) ||
+			    (is_hugetlb_dev(st_buf->st_dev, &hugetlb_flag) && kdat.has_memfd_hugetlb)) {
```

Comment:
Should it be `kdat.has_memfd_hugetlb && is_hugetlb_dev(st_buf->st_dev, &hugetlb_flag)`?

Reply:
@minhbq-99 here we have the following condition: `( is_hugetlb_dev(...) && kdat.has_memfd_hugetlb )`

This condition will be _true_ if both `is_hugetlb_dev()` AND `kdat.has_memfd_hugetlb` are true.
It would be _false_, if `kdat.has_memfd_hugetlb` is false.
If `kdat.has_memfd_hugetlb` is false, we don't need to call `is_hugetlb_dev()`.
Swapping `kdat.has_memfd_hugetlb` and `is_hugetlb_dev(...)` is just an optimization.

Reply:
Thanks, I get your point.